sqldelight-compiler/src/test/kotlin/app/cash/sqldelight/core/queries/AsyncSelectQueryTypeTest.kt

```diff
@@ -0,0 +1,154 @@
+package app.cash.sqldelight.core.queries
+
+import app.cash.sqldelight.core.TestDialect
+import app.cash.sqldelight.core.compiler.SelectQueryGenerator
+import app.cash.sqldelight.core.dialects.intType
+import app.cash.sqldelight.core.dialects.textType
+import app.cash.sqldelight.dialects.postgresql.PostgreSqlDialect
+import app.cash.sqldelight.test.util.FixtureCompiler
+import com.google.common.truth.Truth.assertThat
+import com.squareup.burst.BurstJUnit4
+import org.junit.Assume.assumeTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import org.junit.runner.RunWith
+
+@RunWith(BurstJUnit4::class)
+class AsyncSelectQueryTypeTest {
+  @get:Rule val tempFolder = TemporaryFolder()
+
+  @Test fun `returning clause correctly generates an async query function`(dialect: TestDialect) {
+    assumeTrue(dialect in listOf(TestDialect.POSTGRESQL, TestDialect.SQLITE_3_35))
+    val file = FixtureCompiler.parseSql(
+      """
+      |CREATE TABLE data (
+      |  val1 TEXT,
+      |  val2 TEXT
+      |);
+      |
+      |insertReturning:
+      |INSERT INTO data
+      |VALUES ('sup', 'dude')
+      |RETURNING *;
+      |""".trimMargin(),
+      tempFolder,
+      dialect = PostgreSqlDialect(),
```

Comment:
should use the burst parameter

Reply:
or remove the burst parameter

Reply:
This was copied straight from `SelectQueryTypeTest`. Do you want those ones changed in this PR too?